pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/SinkBase.java

```diff
@@ -0,0 +1,258 @@
+            @ApiResponse(code = 408, message = "Request timeout"),
+            @ApiResponse(code = 404, message = "The function doesn't exist")
+    })
+    @Path("/{tenant}/{namespace}/{sinkName}")
+    public Response getSinkInfo(final @PathParam("tenant") String tenant,
+                                final @PathParam("namespace") String namespace,
+                                final @PathParam("sinkName") String sinkName) throws IOException {
+        return functions.getFunctionInfo(tenant, namespace, sinkName);
+    }
+
+    @GET
+    @ApiOperation(
+            value = "Displays the status of a Pulsar Sink instance",
+            response = FunctionStatus.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 400, message = "Invalid request"),
+            @ApiResponse(code = 403, message = "The requester doesn't have admin permissions"),
+            @ApiResponse(code = 404, message = "The function doesn't exist")
+    })
+    @Path("/{tenant}/{namespace}/{sinkName}/{instanceId}/status")
+    public Response getSinkInstanceStatus(final @PathParam("tenant") String tenant,
+                                          final @PathParam("namespace") String namespace,
+                                          final @PathParam("sinkName") String sinkName,
+                                          final @PathParam("instanceId") String instanceId) throws IOException {
+        return functions.getFunctionInstanceStatus(
+            tenant, namespace, sinkName, instanceId, uri.getRequestUri());
+    }
+
+    @GET
+    @ApiOperation(
+            value = "Displays the status of a Pulsar Sink running in cluster mode",
+            response = FunctionStatus.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 400, message = "Invalid request"),
+            @ApiResponse(code = 403, message = "The requester doesn't have admin permissions")
+    })
+    @Path("/{tenant}/{namespace}/{sinkName}/status")
+    public Response getSinkStatus(final @PathParam("tenant") String tenant,
+                                  final @PathParam("namespace") String namespace,
+                                  final @PathParam("sinkName") String sinkName) throws IOException {
+        return functions.getFunctionStatus(tenant, namespace, sinkName, uri.getRequestUri());
+    }
+
+    @GET
+    @ApiOperation(
+            value = "Lists all Pulsar Sinks currently deployed in a given namespace",
+            response = String.class,
+            responseContainer = "Collection"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 400, message = "Invalid request"),
+            @ApiResponse(code = 403, message = "The requester doesn't have admin permissions")
+    })
+    @Path("/{tenant}/{namespace}")
+    public Response listSinks(final @PathParam("tenant") String tenant,
+                              final @PathParam("namespace") String namespace) {
+        return functions.listFunctions(tenant, namespace);
```

Comment:
This will list all functions, sinks, and sources in the namespace.  For this endpoint to make sense I think we should only return sinks in this namesapce

Reply:
I plan to add these checks in the follow up prs in the coming days.

Reply:
ok